src/geometry/manifold/mfderiv.lean

```diff
@@ -1301,6 +1318,35 @@ lemma mdifferentiable.sub {f : M → E'} (hf : mdifferentiable I 𝓘(𝕜, E')
   (hg : mdifferentiable I 𝓘(𝕜, E') g)  : mdifferentiable I 𝓘(𝕜, E') (f - g) :=
 λ x, (hf x).sub (hg x)
 
+lemma mfderiv_sub (hf : mdifferentiable_at I 𝓘(𝕜, E') f z)
+  (hg : mdifferentiable_at I 𝓘(𝕜, E') g z) :
+  (mfderiv I 𝓘(𝕜, E') (f - g) z : tangent_space I z →L[𝕜] E') =
+  (mfderiv I 𝓘(𝕜, E') f z - mfderiv I 𝓘(𝕜, E') g z : tangent_space I z →L[𝕜] E') :=
+(hf.has_mfderiv_at.sub hg.has_mfderiv_at).mfderiv
+
+end group
+
+section algebra
+variables {z : M} {F' : Type*} [normed_comm_ring F'] [normed_algebra 𝕜 F']
+{f g : M → E'} {p q : M → F'} {I}
+{f' g' : tangent_space I z →L[𝕜] E'}
+{p' q' : tangent_space I z →L[𝕜] F'}
+
+lemma has_mfderiv_at.mul (hp : has_mfderiv_at I 𝓘(𝕜, F') p z p')
+  (hq : has_mfderiv_at I 𝓘(𝕜, F') q z q') :
+  has_mfderiv_at I 𝓘(𝕜, F') (p * q) z (p z • q' + q z • p' : E →L[𝕜] F') :=
+⟨hp.1.mul hq.1, by simpa only with mfld_simps using hp.2.mul hq.2⟩
```

Comment:
You can relax `normed_comm_ring` to `normed_ring` if you use
```suggestion
lemma has_mfderiv_at.mul (hp : has_mfderiv_at I 𝓘(𝕜, F') p z p')
  (hq : has_mfderiv_at I 𝓘(𝕜, F') q z q') :
  has_mfderiv_at I 𝓘(𝕜, F') (p * q) z (p z • q' + mul_opposite.op (q z) • p' : E →L[𝕜] F') :=
⟨hp.1.mul hq.1, by simpa only with mfld_simps using hp.2.mul' hq.2⟩
```

Reply:
I explored what using `mul_opposite.op` looks like a little more in #17802, leaving as an RFC for now. Using `smul_right` instead in a primed lemma would match the existing style for now.

Reply:
Oh, that is nice! I'll use `smul_right` for now.